### DISCUSSION TITLE
IEntity vs IDocument

### DISCUSSION THREAD

**Discussion by MikeKry** [OrchardCMS/OrchardCore#12794] [Q&A]
Hi, 

could you please help me clarify one thing I can't find in docs? 

I wonder, what is the difference between Entity and Document classes (and their interfaces). Just notifced there is also an DocumentEntity..
I realized that Document manager is used when you want single document in DB. In other cases you can use ISession, but is the IEntity interface needed for anything? 

Can I query/create/update IDocument with session? or should I use only document manager?
Is the Entity interface / class needed for creating custom db classes or not?

**Comment by hishamco** [comment]
`IEntity` as the name implies, representing an entity such as person, car .. etc. in OC we use it for built-in entities such as site, activity .. etc. On other hand `IDocument` represents a document of things so you can have templates document, translations document .. etc

Documents has store and could be retreived from database or cache via `DocumentManager`. I might ask @jtkech for more information coz he is the guru for many infrastructure related things in OC ;)

**Comment by ns8482e** [comment]
@MikeKry technically both same represents a json document having dictionary of properties 

However there is major difference in usage, if your need is to store json document independently as multiple record and you want to  query using index then use IEntity. example is Users

If your need is to store as single record and save all child records inside same document separated by a key then use Document. Example is SiteSetting, Templates
